ui/hooks/experiences/useRamps.ts

```diff
@@ -13,13 +18,23 @@ const portfolioUrl = process.env.PORTFOLIO_URL;
 
 const useRamps = (): IUseRamps => {
   const chainId = useSelector(getCurrentChainId);
+  const metaMetricsId = useSelector(getMetaMetricsId);
+  const selectedAddress = useSelector(getSelectedAddress);
 
   const getBuyURI = useCallback((_chainId: Hex) => {
     switch (_chainId) {
       case CHAIN_IDS.SEPOLIA:
         return 'https://faucet.sepolia.dev/';
-      default:
-        return `${portfolioUrl}/buy?metamaskEntry=ext_buy_button`;
+      default: {
+        const params = new URLSearchParams();
+        params.set('metamaskEntry', 'ext_buy_button');
+        params.set('address', selectedAddress);
```

Comment:
I noticed that this wasn't included previously. Is this change related to adding networks? It's not clear why this is now needed

Reply:
We used this pattern with the [previous on-ramp with all providers](https://github.com/MetaMask/metamask-extension/blob/eee4cc3100f02282e7e008b92d4a503284aa48ce/app/scripts/lib/buy-url.js#L68) and we are bringing it back now. We've seen Portfolio conversion data showed a significant drop-off for Extension users due to the connect wallet friction.

Reply:
Perhaps we could consider this addition in a separate PR? I'm aware of that problem, but it seems distinct from the addition of new supported networks, and there are other solutions under consideration.